src/VisualStudio/Core/Def/ProjectSystem/IOpenTextBufferEventListener.cs

```diff
@@ -8,18 +8,20 @@
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
 {
     /// <summary>
-    /// Defines the methods that get called by the <see cref="RunningDocumentTableEventTracker"/>
+    /// Defines the methods that get called by the <see cref="OpenTextBufferProvider"/>
     /// for getting notified about running document table events.
     /// </summary>
-    internal interface IRunningDocumentTableEventListener
+    internal interface IOpenTextBufferEventListener
     {
         /// <summary>
         /// Triggered when a document is opened.
         /// </summary>
         /// <param name="moniker">the moniker of the opened document.</param>
         /// <param name="textBuffer">the text buffer of the opened document)</param>
         /// <param name="hierarchy">the hierarchy of the text buffer if available.</param>
-        void OnOpenDocument(string moniker, ITextBuffer textBuffer, IVsHierarchy? hierarchy, IVsWindowFrame? windowFrame);
+        void OnOpenDocument(string moniker, ITextBuffer textBuffer, IVsHierarchy? hierarchy);
+
+        void OnDocumentOpenedIntoWindowFrame(string moniker, IVsWindowFrame windowFrame);
```

Comment:
I think previously we'd fire OnOpenDocument multiple times, sometimes with windowFrame being null early on, and then a later call it being non-null. The semantics weren't clear, because in some cases we don't have a window frame. Since only one piece of code actually cared about the window frame, this just breaks out that hookup into a separate event.